test/modm/architecture/interface/i2c_test.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020, Sascha Schade
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include "i2c_test.hpp"
+
+#include <modm/architecture/interface/i2c.hpp>
+#include <modm-test/mock/iodevice.hpp>
+
+// ----------------------------------------------------------------------------
+static modm_test::platform::IODevice device;
+
+void
+I2cTest::setUp()
+{
+	device.clear();
+	stream = new modm::IOStream(device);
+}
+
+void
+I2cTest::tearDown()
+{
+	delete stream;
```

Comment:
Explicit new/delete should be avoided in modern C++. If `stream` has to be pointer, at least `std::unique_ptr` should be used which takes care of deleting the object exactly once but probably we don't need a pointer here at all.

Reply:
These test classes are statically allocated, so their destructor is never called, so `std::unique_ptr` wouldn't help here.
Also note that setUp() and tearDown() are called before and after every single member function test case.

We do this to not have to allocate all RAM for all tests at once, since only one can test runs at any time anyways.

Reply:
You could still use `std::unique_ptr` in the places you have new/delete here. `stream = std::make_unique<modm::IOStream>(device);` and `stream = nullptr`. For sure you could argue what the benefit for this isolated example is. If you prefer `new` / `delete` here go with it.